packages/nimble-components/src/table-column/number-text/tests/table-column-number-text.spec.ts

```diff
@@ -0,0 +1,406 @@
+                    async () => {
+                        columnInstances.column1.format = format;
+                        await connect();
+
+                        await element.setData([{ number1: testCase.value }]);
+                        await waitForUpdatesAsync();
+
+                        expect(pageObject.getRenderedCellContent(0, 0)).toBe(
+                            testCase.expectedRenderedString
+                        );
+                        expect(
+                            pageObject.getRenderedGroupHeaderContent(0)
+                        ).toContain(testCase.expectedRenderedString);
+                    }
+                );
+            }
+        }
+    });
+
+    describe('with default formatting', () => {
+        const testCases: readonly NumericTestCase[] = [
+            {
+                name: '"E" renders as "+e"',
+                value: 28729375089724643,
+                expectedRenderedString: '2.872937508972464e+16'
+            },
+            {
+                name: '"-E" renders as "-e"',
+                value: 0.0000002358967325,
+                expectedRenderedString: '2.358967325e-7'
+            },
+            {
+                name: 'displays at most 16 decimal places',
+                value: 1.234567890123456789,
+                expectedRenderedString: '1.234567890123457'
+            },
+            {
+                name: 'shows less than 16 decimal places for numbers without many decimal',
+                value: 1.23,
+                expectedRenderedString: '1.23'
+            }
+        ] as const;
+
+        const focused: string[] = [];
+        const disabled: string[] = [];
+        for (const testCase of testCases) {
+            const specType = getSpecTypeByNamedList(
+                testCase,
+                focused,
+                disabled
+            );
+            // eslint-disable-next-line @typescript-eslint/no-loop-func
+            specType(
+                `${testCase.name}`,
+                // eslint-disable-next-line @typescript-eslint/no-loop-func
+                async () => {
+                    await connect();
+
+                    await element.setData([{ number1: testCase.value }]);
+                    await waitForUpdatesAsync();
+
+                    expect(pageObject.getRenderedCellContent(0, 0)).toBe(
+                        testCase.expectedRenderedString
+                    );
+                    expect(
+                        pageObject.getRenderedGroupHeaderContent(0)
+                    ).toContain(testCase.expectedRenderedString);
+                }
+            );
+        }
+    });
+
+    describe('with integer formatting', () => {
+        beforeEach(async () => {
+            columnInstances.column1.format = NumberTextFormat.integer;
+            await waitForUpdatesAsync();
+        });
+
+        const testCases: readonly NumericTestCase[] = [
```

Comment:
We should probably have the similar (round-down and round-up) test cases for negative numbers too. Intl.NumberFormat` has configuration for that (`roundingMode`), though the default behavior looks fine.